runelite-client/src/main/java/net/runelite/client/plugins/interfacestyles/InterfaceStylesPlugin.java

```diff
@@ -129,6 +136,48 @@ public void onConfigChanged(ConfigChanged config)
 		}
 	}
 
+	@Subscribe(priority = 9)
+	public void onMenuOpened(MenuOpened event)
+	{
+		condensePlayerOptions();
+	}
+
+	private void condensePlayerOptions()
+	{
+		if (!config.condensePlayerOptions())
+		{
+			return;
+		}
+
+		MenuEntry[] menuEntries = client.getMenuEntries();
+		List<MenuEntry> menuEntryList = new ArrayList<>();
+		MenuEntry playerSubmenu = null;
+		for (int i = menuEntries.length - 1; i >= 0; i--)
+		{
+			MenuEntry rawMenuEntry = menuEntries[i];
+
+			Player player = rawMenuEntry.getPlayer();
+			if (player != null && !Objects.equals(rawMenuEntry.getOption(), "Use") && !Objects.equals(
+					rawMenuEntry.getOption(), "Cast"))
+			{
+				if (playerSubmenu == null || !Objects.equals(playerSubmenu.getTarget(), rawMenuEntry.getTarget()))
+				{
+					playerSubmenu = client.createMenuEntry(0)
+							.setIdentifier(rawMenuEntry.getIdentifier()).setTarget(rawMenuEntry.getTarget())
+							.setType(MenuAction.RUNELITE_SUBMENU).setParam0(rawMenuEntry.getParam0())
+							.setParam1(rawMenuEntry.getParam1());
+					menuEntryList.add(0, playerSubmenu);
```

Comment:
ArrayList.add(0) is quite inefficient since it has to shift the entire list for each insert. I think you can just reverse iteration instead and then add() to the end?